File: files/CH-144.py
import pandas as pd

path = "CH-144 First transaction in each month.xlsx"
input = pd.read_excel(path, usecols="C:E", skiprows=1, nrows=26)
test = pd.read_excel(path, usecols="G:I", skiprows=1, nrows=4).rename(columns=lambda x: x.split('.')[0])

result = input.assign(month=input['Date'].dt.month).groupby('month').head(1).drop(columns='month').reset_index(drop=True)

print(result.equals(test))  # True

File: files/100-199/144/CH-144.R
library(tidyverse)
library(readxl)

path = "files/CH-144 First transaction in each month.xlsx"
input = read_excel(path, range = "C2:E27")
test  = read_excel(path, range = "G2:I6")

result = input %>%
  mutate(month = month(Date)) %>%
  filter(row_number() == 1, .by = month) %>%
  select(-month)

all.equal(result, test, check.attributes = FALSE)
# [1] TRUE